### PR TITLE
WIP: Search Near... frontend work

### PR DESCRIPTION
**Pull request by weedySeaDragon**
### PT Story:  Find Companies Near Me
https://www.pivotaltracker.com/story/show/140181097

**_New and Improved version 2.0!  See the first screenshot below_**

### WIP Notes

- I changed from using the _check box_ to using a _button._   
   - the image for the button is just a placeholder.  Obviously we need a better one!
   - it would be great to use a font-awesome icon, but I couldn't figure out how to do it

 **Order matters**: If you _first_ choose a search filter (ex: kategori/category), _then_ click 'Search Near Me', the results are correct and what you'd expect.
   But if you first click 'Search Near Me' and then enter search filters, the 'Search Near Me' button results are thrown away.
  - This behavior is difficult to change because of how rigid the Ransack gem is and because it changes the ActiveRecord queries.
  - **_How can we let the users know this is how it works?_**

- There is now a box at the bottom where you can enter your fake location.  This is needed for development and testing. 
  - I still need to try to add a button to that fake box so that we can also test 'browser won't provide the user location'

**~~What to display if no companies are found near the user?~~**
   - ~~I thought all I need to do was display a flash message with the controller, but that's not working.~~

**What should the default distance be?  I have it set to 20km for now.**

- had to change the parameters passed to the controller
  - I need to review the controller specs to see if additional tests are needed or existing tests need to be changed.  If anyone sees specs needed or modifications needed, please do comment.

**Integration testing:**  Not sure how to accomplish some of this:
   When Google renders a map, it does _not_ render the element id for the controls that are added (e.g. the "search near me" checkbox).  So I'm not sure if or how to check for that.
   We can certainly check the results, but I'm not sure if we'll be able to check the initial state (e.g. that the "search near me" checkbox has been checked or not).

**Etc:**

- I didn't spend much time styling the checkbox.  Suggestions needed!

- I couldn't stand watching the long and slow queries that are generated by the current `Company.with_members` scope.  So I created the faster, more efficient scope (and resulting query).  But since I have not tested it, I will revert to the original, tested scope and submit a PivotalTracker task for it.  (I just haven't cleaned up the code yet.)

- I removed the "Show/Hide Search Box" button.  since it's not needed with this layout.  But this changed a bunch of feature tests.  So I added some steps and changed the features.  **I know that one feature is failing right now.** It seems like an AJAX/timing thing, but haven't had time to look at it.

### Changes proposed in this pull request:
1. Changed the parameters passed to the controller 
2. Changed the layout for the search filters and map to vertical
3. Removed the 'show/hide search box' button. Changed the features to search for results only in the companies list

#### version 2: uses a button instead of a checkbox; area to enter 'fake' current location

search near me v2 2018-11-27 at 111631 pm

---

#### Vertical Layout for Search Filters and Map

_all companies are being shown_

search-near-me-v2 show all

---

#### Show all companies with the "Dagis" kategori/category

_Dagis companies everywhere_ ( = 3)

search-near-me-v2 all dagis kategori

---

#### Show all "Dagis" companies within 100km of the user's location (= Stockholm)

_There are 2 Dagis companies within 100km of Stockholm_

search-near-me-v2 dagis 100km stockholm

---

### Ready for review:

@AgileVentures/shf-project-team 
@patmbolger 
@thesuss

### CODE REVIEW

File: app/assets/stylesheets/maps.scss [hunk 1]
```diff
@@ -1,38 +1,36 @@
...
+  height: 405px; // doesn't work with 100% TODO: how does this look on small screens? may need to fix for those
+  width: 100%;
 
+  #search-near-me {
+    background-color: white;
+    border: 1.5pt solid $checkbox-border-color;
+    padding-bottom: 3px;
+    padding-left: 4px;
+
+    #search-near-me-checkbox {
```
[the remark this answers is not in the document]
Comment (weedySeaDragon) [reply app/assets/stylesheets/maps.scss:60]: ignore this

**Comment by patmbolger** [comment]
Here are my first comments based on just fiddling with the UI:

1. The "search near me" selection mechanism is very laborious to go through monotonic changes in the counter.  How about discrete selections like 5, 10, 25, 50, 100, 200, Any   ?

2. Every time the page is refreshed (full or partial) it seems the "near" radius gets reset back to 20.

3. I think the idea of having 2 different search triggers (buttons) is _very_ confusing, and we'll never get the user to understand the difference, and the expected sequence of search actions that must occur (the idea of the search (form) criteria being set first, then the "near" button being clicked (and the opposite won't work) just won't work, IMO.

4) Continuing with the above thought, I think we need _one_ search button (that applies to any textual search criteria _and_ the "near me" radius (if specified).  The "Search near me" really should be a checkbox.

5) If I search on region, I see this alert:

screen shot 2018-11-28 at 3 49 15 pm

followed by a refreshed page that had dropped the map:

screen shot 2018-11-28 at 3 50 07 pm

If I reload the browser page I keep seeing the view above.  This persists until I clear the region criteria and click "Search" again.

6) The same phenomena as above happens for "Municipality".

7) The alert might be clearer if it said: `Sorry. No companies matching your search criteria were found near you.`

**Comment by weedySeaDragon** [comment]
I totally agree with your comments, @patmbolger.  Having one button to push to do the search is the natural and expected UX.

Having Ransack in there adds a whole layer of difficulty since it is so tightly and rigidly bound to that search form.  I'll ponder ways to work with it. I'm afraid we'll have to re-write the search form view.

 Any insights would be most appreciated. :-)

**Comment by weedySeaDragon** [comment]
Closing this:  What we really needed was "Show the companies near me" -- accomplished in PR #572 .   This feature: search for companies near me using the map -- can come later.

**Pull request closed without merging**